src/main/java/nextstep/subway/favorite/application/FavoriteService.java

```diff
@@ -0,0 +1,65 @@
+package nextstep.subway.favorite.application;
+
+import nextstep.subway.favorite.domain.Favorite;
+import nextstep.subway.favorite.domain.FavoriteRepository;
+import nextstep.subway.favorite.dto.FavoriteRequest;
+import nextstep.subway.favorite.dto.FavoriteResponse;
+import nextstep.subway.member.application.MemberService;
+import nextstep.subway.member.domain.Member;
+import nextstep.subway.station.application.StationService;
+import nextstep.subway.station.domain.Station;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class FavoriteService {
+    private final FavoriteRepository favoriteRepository;
+    private final MemberService memberService;
+    private final StationService stationService;
+
+    public FavoriteService(FavoriteRepository favoriteRepository, MemberService memberService, StationService stationService) {
+        this.favoriteRepository = favoriteRepository;
+        this.memberService = memberService;
+        this.stationService = stationService;
+    }
+
+    @Transactional
+    public FavoriteResponse saveFavorite(Long memberId, FavoriteRequest request) {
+        Member member = memberService.findMemberById(memberId);
+        Station source = stationService.findById(request.getSourceId());
+        Station target = stationService.findById(request.getTargetId());
+        Favorite favorite = favoriteRepository.save(new Favorite(member, source, target));
+
+        return FavoriteResponse.of(favorite);
+    }
+
+    @Transactional(readOnly = true)
+    public List<FavoriteResponse> findAllFavoritesByMemberId(Long memberId) {
+        List<Favorite> favorites = favoriteRepository.findAllByMemberId(memberId);
+
+        return favorites.stream()
+                .map(FavoriteResponse::of)
+                .collect(Collectors.toList());
+    }
+
+    @Transactional
+    public void deleteFavoriteById(Long id) {
+        try {
+            favoriteRepository.deleteById(id);
+        }catch (Exception e) {
```

Comment:
```suggestion
        } catch (Exception e) {
```

매의 눈 @_@

IntelliJ Auto Formatting 기능을 사용하는 습관을 들여보아요.

The auto-formatting shortcut in IntelliJ

- For windows Ctrl + Alt + L.
- For ubuntu Ctrl + Alt + windows + L.
- For Mac Command + Option + L.

Reply:
코드 작성 중에 자주 사용하도록 습관을 들여보겠습니다 :)